Not including '?' or 'or {...}' in beginning of match statement produces a C error.

**V version: V 0.2.2 60bc280.b15156d** 
**OS: linux, Linux Mint 20.1 (the Ubuntu based version; not Debian based)** 

**What did you do?**
Called a function that returns an option in between `match` and `{` but forgot to place the `?` at the end of the function call to signify that I wanted the option to be sent to the parent function. I was able to repeat this error with this code while trying to figure out what I did wrong in my project.
```v
enum Test {
    good = 1
    ok = 0
    bad = -1
}

fn str_to_enum(str string) ?Test {
    if str == 'good' { return .good }
    else if str == 'ok' { return .ok }
    else if str == 'bad' { return .bad }
    else {
        return error('Invalid value for enum')
    }
}

fn main() {
    str := 'good'
    // this function call should be followed by a '?'
    // the C error is also produced here 
    match str_to_enum(str) { 
        .good { println('Good') }
        else { println('Not good') }
    }
}
```

**What did you expect to see?**
I expected to see  this error:
```error: str_to_enum() returns an option, so it should have either an `or {}` block, or `?` at the end```
 
**What did you see instead?**
Instead of the above I got this C error:
```
/tmp/v/test.3366003708170568310.tmp.c:9435: error: invalid operand types for binary operation
/tmp/v/test.3366003708170568310.tmp.c:9435: error: invalid aggregate type for register load
builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -5135,6 +5135,7 @@ pub fn (mut c Checker) match_expr(mut node ast.MatchExpr) ast.Type {
 	// since it is used in c.match_exprs() it saves checking twice
 	node.cond_type = c.table.mktyp(cond_type)
 	c.ensure_type_exists(node.cond_type, node.pos) or { return ast.void_type }
+	c.check_expr_opt_call(node.cond, cond_type)
 	cond_type_sym := c.table.get_type_symbol(cond_type)
 	if cond_type_sym.kind !in [.interface_, .sum_type] {
 		node.is_sum_type = false

--- a/vlib/v/checker/tests/optional_fn_err.out
+++ b/vlib/v/checker/tests/optional_fn_err.out
@@ -151,10 +151,18 @@ vlib/v/checker/tests/optional_fn_err.vv:69:18: error: bar() returns an option, s
    69 |     println(arr.any(bar(true)))
       |                     ~~~~~~~~~
    70 |     println(arr.all(bar(true)))
-   71 | }
+   71 |
 vlib/v/checker/tests/optional_fn_err.vv:70:18: error: bar() returns an option, so it should have either an `or {}` block, or `?` at the end
    68 |     println(arr.filter(bar(true)))
    69 |     println(arr.any(bar(true)))
    70 |     println(arr.all(bar(true)))
       |                     ~~~~~~~~~
-   71 | }
+   71 | 
+   72 |     match bar(0) {
+vlib/v/checker/tests/optional_fn_err.vv:72:8: error: bar() returns an option, so it should have either an `or {}` block, or `?` at the end
+   70 |     println(arr.all(bar(true)))
+   71 | 
+   72 |     match bar(0) {
+      |           ~~~~~~
+   73 |         0 { }
+   74 |         else { }

--- a/vlib/v/checker/tests/optional_fn_err.vv
+++ b/vlib/v/checker/tests/optional_fn_err.vv
@@ -68,4 +68,9 @@ fn main() {
 	println(arr.filter(bar(true)))
 	println(arr.any(bar(true)))
 	println(arr.all(bar(true)))
+
+	match bar(0) {
+		0 { }
+		else { }
+	}
 }

